src/main/java/com/sprint/mission/discodeit/controller/SseController.java

```diff
@@ -0,0 +1,46 @@
+package com.sprint.mission.discodeit.controller;
+
+import com.sprint.mission.discodeit.security.DiscodeitUserDetails;
+import com.sprint.mission.discodeit.service.SseService;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+
+@RestController
+@RequestMapping("/api")
+@RequiredArgsConstructor
+public class SseController {
+
+  private final SseService sseService;
+
+  // SSE 연결 엔드포인트
+  // 클라이언트가 GET 요청으로 /api/sse 에 접속하면 SSE 연결이 시작됨
+  @GetMapping(value = "/sse", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
+  public ResponseEntity<SseEmitter> streamData(
+      @RequestHeader(value = "Last-Event-ID", required = false) String lastEventId) {
```

Comment:
`@AuthenticationPrincipal` 애노테이션을 이용하면 컨트롤러 단에서 getCurrentUserId로 인증정보를 확인하는 코드를 제거할 수 있을 것 같아요.

```suggestion
  public ResponseEntity<SseEmitter> streamData(
      @AuthenticationPrincipal DiscodeitUserDetails userDetails,
      @RequestHeader(value = "Last-Event-ID", required = false) String lastEventId) {
```